### DISCUSSION TITLE
Got 404 error when access by proxy.

### DISCUSSION THREAD

**Discussion by ysjjovo** [aws-solutions/centralized-logging-with-opensearch#99] [Q&A]
1. Enable access proxy
2. Add a CNAME record,map 'log.ysjjovo.ml' to  access proxy ALB address.
3. Use Chrome to access 'log.ysjjovo.ml'
4. It returns a 404 page
5. Login in  to the nginx server.here is the /etc/nginx/conf.d/default.conf
```
server {
    listen 443;
    server_name log.ysjjovo.ml;
    resolver 172.16.0.2 [::1]:5353 valid=30s;
    rewrite ^/$ https://log.ysjjovo.ml/_dashboards  redirect;
    ssl_certificate /etc/nginx/cert.crt;
    ssl_certificate_key /etc/nginx/cert.key;
    ssl on;
    ssl_session_cache builtin:1000 shared:SSL:10m;
    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
    ssl_ciphers HIGH:!aNULL:!eNULL:!EXPORT:!CAMELLIA:!DES:!MD5:!PSK:!RC4;
    ssl_prefer_server_ciphers on;
    set $es_endpoint vpc-log-gxiq4wzyjl3mghkexxfe2lcs6i.us-east-2.es.amazonaws.com;
    set $cognito_endpoint loghub-portal-260527533511.auth.us-east-2.amazoncognito.com;
    location ^~ /_dashboards {
        # Forward requests to Kibana
        proxy_pass https://$es_endpoint;
        # Handle redirects to Amazon Cognito
        proxy_redirect https://$cognito_endpoint  https://log.ysjjovo.ml;
        # Update cookie domain and path
        proxy_cookie_domain $es_endpoint log.ysjjovo.ml;
        # Response buffer settings
        proxy_buffer_size 128k;
        proxy_buffers 4 256k;
        proxy_busy_buffers_size 256k;
        # Ignore client disconnection
        proxy_ignore_client_abort  on;
    }
    location ~ \/(log|sign|error|fav|forgot|change|confirm|mfa) {
        # Forward requests to Cognito
        proxy_pass https://$cognito_endpoint;
        # Handle redirects to Kibana
       # proxy_redirect https://$es_endpoint  https://log.ysjjovo.ml;
        # Handle redirects to Amazon Cognito
       # proxy_redirect https://$cognito_endpoint  https://log.ysjjovo.ml;
        # Update cookie domain
        proxy_cookie_domain $cognito_endpoint log.ysjjovo.ml;
        # Ignore client disconnection
        proxy_ignore_client_abort  on;
        internal;
    }
}
```
6. comment proxy_redirect from location '_dashboards'
```#proxy_redirect https://$cognito_endpoint  https://log.ysjjovo.ml;```
7. Save and quit,then run command ```nginx -s reload``` to take effect
8. Use Chrome to access 'log.ysjjovo.ml'
9. It redirect to Congnito
10. Login with username and password,redirect to a inner address 'pc-log-gxiq4wzyjl3mghkexxfe2lcs6i.us-east-2.es.amazonaws.com',and I cannot access it.

**Comment by Paul-AWS** [comment]
By default, you can use "https://log.ysjjovo.ml/_dashboards/" to visit your OpenSearch Dashboard. Please try to restore the nginx settings, and use the URL above to visit the OpenSearch Dashboard.
You can also find the URL in the Centralized Logging with OpenSearch console. ("Clusters" in the left sidebar -> "OpenSearch Domains" -> YOUR DOMAIN -> "General configuration" -> "Access Proxy" -> "Link")

**Reply by ysjjovo** [reply]
Visit URL ```https://log.ysjjovo.ml/``` will redirect to URL ```https://log.ysjjovo.ml/_dashboards ```,due to this directive ```rewrite ^/$ https://log.ysjjovo.ml/_dashboards  redirect;```.It's the same result of getting 404 ERROR.
